database/migrations/mikoyan/src/main.rs

```diff
@@ -0,0 +1,165 @@
+    } else {
+        num_docs_in_collection / num_threads
+    };
+
+    let m = MultiProgress::new();
+    for thread_id in 0..num_threads {
+        let num_docs_to_handle = if thread_id == num_threads - 1 {
+            // Handle any remainder
+            num_docs_per_thread + num_docs_in_collection % num_threads
+        } else {
+            num_docs_per_thread
+        };
+
+        println!("Thread {}: {:?}", thread_id, num_docs_to_handle);
+
+        let args = args.clone();
+
+        let m_clone = m.clone();
+        let handle: JoinHandle<Result<(), mongodb::error::Error>> = tokio::spawn(async move {
+            match connect_and_process(args, num_docs_to_handle, thread_id, m_clone).await {
+                Ok(_) => Ok(()),
+                Err(e) => Err(e),
+            }
+        });
+
+        handles.push(handle);
+    }
+
+    let mut file = tokio::fs::OpenOptions::new()
+        .append(true)
+        .create(true)
+        .open(args.logs)
+        .await?;
+
+    for handle in handles {
+        if let Err(e) = handle.await {
+            // Write errors to logs file
+            file.write_all(format!("{}\n", e).as_bytes()).await?;
+        }
+    }
+    Ok(())
+}
+
+async fn connect_and_process(
+    args: Args,
+    num_docs_to_handle: usize,
+    thread_id: usize,
+    m: MultiProgress,
+) -> Result<(), mongodb::error::Error> {
+    let collection = get_collection(&args.uri).await?;
+
+    let find_ops = FindOptions::builder()
+        .limit(num_docs_to_handle as i64)
+        .skip((thread_id * num_docs_to_handle) as u64)
+        .projection(doc! {
+            "_id": 1,
+            "completedChallenges": 1,
+            "progressTimestamps": 1,
+            "partiallyCompletedChallenges": 1,
+            "yearsTopContributor": 1,
+            "savedChallenges": 1,
+            "badges": 1,
+        })
+        .build();
+    let mut cursor = collection.find(doc! {}, find_ops).await?;
+
+    let sty = ProgressStyle::with_template(
+        "[{elapsed_precise}] {bar:40.cyan/blue} {pos:>7}/{len:7} {msg}",
+    )
+    .unwrap()
+    .progress_chars("##-");
+
+    let pb = m.add(ProgressBar::new(num_docs_to_handle as u64));
+    pb.set_style(sty);
+
+    let mut logs_file = tokio::fs::OpenOptions::new()
+        .append(true)
+        .create(true)
+        .open(args.logs)
+        .await?;
+
+    let mut count: usize = 0;
+    let epoch = (num_docs_to_handle / 1000).max(1);
+    while let Some(user) = cursor.try_next().await? {
+        match normalize_user(&user) {
+            Ok(normalized_user) => {
+                // _id exists, because `normalize_user` returns an error if it does not
+                let id = user.get_object_id("_id").unwrap();
+                let filter = doc! {"_id": id};
+                let update_options = UpdateOptions::builder()
+                    .array_filters(vec![doc! {
+                        "el": {"$exists": true},
+                    }])
+                    .build();
```

Comment:
What does this do?

Reply:
Docs: https://www.mongodb.com/docs/manual/reference/operator/update/positional-filtered/

`normalize.rs` has this:

```rust
"completedChallenges.$[el].__cachedRelations": "",
"completedChallenges.$[el].__data": "",
"completedChallenges.$[el].__dataSource": "",
"completedChallenges.$[el].__persisted": "",
"completedChallenges.$[el].__strict": "",
```

So this defines `el`, and prevents errors by only performing the `$[el]` update when it exists.